test/test-parse-warning.cpp

```diff
@@ -737,8 +737,36 @@ TEST_F(test_parse_warning, warn_on_pointless_nullish_coalescing_operator) {
     p.parse_and_visit_expression();
   }
 }
-}
-}
+
+TEST_F(test_parse_warning, warn_on_variable_assigned_to_self_is_noop) {
```

Comment:
:+1:

Reply:
Added checks to the test:

- Assert `x = \u{78};` warns
- Assert `(x) = x;` warns
- Assert `x = ((x));` warns